ada_feeding_perception/ada_feeding_perception/face_detection.py

```diff
@@ -0,0 +1,314 @@
+        """
+        Read the parameters for this node.
+
+        Returns
+        -------
+        face_model_name: The name of the face detection model checkpoint to use
+        face_model_base_url: The URL to download the model checkpoint from if it is not already downloaded
+        landmark_model_name: The name of the facial landmark detection model checkpoint to use
+        landmark_model_base_url: The URL to download the model checkpoint from if it is not already downloaded
+        model_dir: The location of the directory where the model checkpoint is / should be stored
+        """
+        return self.declare_parameters(
+            "",
+            [
+                (
+                    "face_model_name",
+                    None,
+                    ParameterDescriptor(
+                        name="face_model_name",
+                        type=ParameterType.PARAMETER_STRING,
+                        description="The name of the face detection model checkpoint to use",
+                        read_only=True,
+                    ),
+                ),
+                (
+                    "face_model_base_url",
+                    None,
+                    ParameterDescriptor(
+                        name="face_model_base_url",
+                        type=ParameterType.PARAMETER_STRING,
+                        description="The URL to download the model checkpoint from if it is not already downloaded",
+                        read_only=True,
+                    ),
+                ),
+                (
+                    "landmark_model_name",
+                    None,
+                    ParameterDescriptor(
+                        name="landmark_model_name",
+                        type=ParameterType.PARAMETER_STRING,
+                        description="The name of the face detection model checkpoint to use",
+                        read_only=True,
+                    ),
+                ),
+                (
+                    "landmark_model_base_url",
+                    None,
+                    ParameterDescriptor(
+                        name="landmark_model_base_url",
+                        type=ParameterType.PARAMETER_STRING,
+                        description="The URL to download the model checkpoint from if it is not already downloaded",
+                        read_only=True,
+                    ),
+                ),
+                (
+                    "model_dir",
+                    None,
+                    ParameterDescriptor(
+                        name="model_dir",
+                        type=ParameterType.PARAMETER_STRING,
+                        description="The location of the directory where the model checkpoint is / should be stored",
+                        read_only=True,
+                    ),
+                ),
+            ],
+        )
+
+    def toggle_face_detection_callback(self, request, response):
```

Comment:
Please add type annotations here. I think it should be:

`def toggle_face_detection_callback(self, request: ToggleFaceDetection.Request, response: ToggleFaceDetection.Response) -> ToggleFaceDetection.Response:`